nomos-services/time/src/backends/ntp/async_client.rs

```diff
@@ -63,3 +87,39 @@ impl AsyncNTPClient {
             .map_err(Error::Sntp)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{
+        net::{IpAddr, Ipv4Addr, SocketAddr},
+        time::Duration,
+    };
+
+    use super::*;
+
+    // This test is disabled macOS because NTP v4 requests fail on Github's macOS
+    // runners. Everywhere else it works fine, including other macOS machines.
+    // The request seems to be sent successfully, but the test timeouts when
+    // receiving the response. Responses only come through when querying
+    // `time.windows.com`, which runs NTP v3. The library we're using,
+    // [`sntpc`], requires NTP v4.
+    #[cfg(not(target_os = "macos"))]
```

Comment:
It seems that `time.google.com` fails on `ubuntu-latest` github runner (but works well on our self-hosted runner) so it seems that it's not only macos related. I've seen that there are tests with fake ntp server, do you think those would be enough? And just curious, maybe you tried this with `[self-hosted, macOS]` runner?

Reply:
These are running with `self-hosted` and `macOS`, I didn't test it with `ubuntu` hence I didn't notice.
Where are we running tests against `ubuntu-latest`?